pkg/ddevapp/traefik.go

```diff
@@ -394,19 +402,113 @@ func configureTraefikForApp(app *DdevApp) error {
 	if !sigExists {
 		util.Debug("Not creating %s because it exists and is managed by user", traefikYamlFile)
 	} else {
-		f, err := os.Create(traefikYamlFile)
+
+		//create a temp file to hold the base traefik config while we merge and process it with the dynamic_config.*.yaml files
+		dynamicConfigTemp, err := os.CreateTemp("", "dynamic_config-")
 		if err != nil {
-			return fmt.Errorf("failed to create Traefik config file: %v", err)
+			return err
 		}
+
 		t, err := template.New("traefik_config_template.yaml").Funcs(getTemplateFuncMap()).ParseFS(bundledAssets, "traefik_config_template.yaml")
 		if err != nil {
 			return fmt.Errorf("could not create template from traefik_config_template.yaml: %v", err)
 		}
 
-		err = t.Execute(f, templateData)
+		err = t.Execute(dynamicConfigTemp, templateData)
```

Comment:
We don't need to do this whole templating stage do we, since we template the results of the merge?

Reply:
You're already creating a temp file with the contents if I'm not mistaken.

Reply:
If it's not valid yaml with the templates in it, then you're right. But how is that going to work below, where you have the same problem? All of the files would have to be preprocessed in that case.